src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -10,481 +10,42 @@
-
-        Map cinvDocMap = new ReflectionMap(document);
-        Map<String, Object> parameterMap = new FallbackMap<String, Object>(cinvDocMap);
-
-        Map<String, Object> primaryKeys = new HashMap<>();
-        primaryKeys.put(KFSPropertyConstants.UNIVERSITY_FISCAL_YEAR,
-                document.getAccountingPeriod().getUniversityFiscalYear());
-        primaryKeys.put(KFSPropertyConstants.PROCESSING_CHART_OF_ACCT_CD,
-                document.getAccountsReceivableDocumentHeader().getProcessingChartOfAccountCode());
-        primaryKeys.put(KFSPropertyConstants.PROCESSING_ORGANIZATION_CODE,
-                document.getAccountsReceivableDocumentHeader().getProcessingOrganizationCode());
-        SystemInformation sysInfo = businessObjectService.findByPrimaryKey(SystemInformation.class, primaryKeys);
-        parameterMap.put(KFSPropertyConstants.DOCUMENT_NUMBER, document.getDocumentNumber());
-        if (ObjectUtils.isNotNull(document.getDocumentHeader().getWorkflowDocument().getDateCreated())) {
-            parameterMap.put(KFSPropertyConstants.DATE, getDateTimeService()
-                    .toDateString(document.getDocumentHeader().getWorkflowDocument().getDateCreated().toDate()));
-        }
-        
-        setFinalStatusDate(document, parameterMap);
+        Map<String, String> templateParameters = super.getTemplateParameterList(document);
         
-        parameterMap.put(KFSPropertyConstants.PROPOSAL_NUMBER, document.getInvoiceGeneralDetail().getProposalNumber());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.NAME,
-                document.getBillingAddressName());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.ADDRESS_LINE1,
-                document.getBillingLine1StreetAddress());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.ADDRESS_LINE2,
-                document.getBillingLine2StreetAddress());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.CITY, document.getBillingCityName());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.STATE, document.getBillingStateCode());
-        parameterMap.put(KFSPropertyConstants.PAYEE + "." + KFSPropertyConstants.ZIPCODE, document.getBillingZipCode());
-        parameterMap.put(ArPropertyConstants.ADVANCE_FLAG,
-                ArConstants.BillingFrequencyValues.isPredeterminedBilling(document.getInvoiceGeneralDetail()));
-        parameterMap.put(ArPropertyConstants.REIMBURSEMENT_FLAG,
-                !ArConstants.BillingFrequencyValues.isPredeterminedBilling(document.getInvoiceGeneralDetail()));
-        parameterMap.put(
-                ArPropertyConstants.ACCOUNT_DETAILS + "." + KFSPropertyConstants.CONTRACT_CONTROL_ACCOUNT_NUMBER,
-                getRecipientAccountNumber(document.getAccountDetails()));
-        
-        if (document.getInvoiceGeneralDetail().isFinalBillIndicator()) {
```

Comment:
This "if-else" block needs to be preserved, since it contains our customization for adding the FINAL_BILL and PARTIAL_BILL indicators to the PDF.

Reply:
good catch, thank you